pennylane/qnodes/base.py

```diff
@@ -336,13 +489,9 @@ def _construct(self, args, kwargs):
                 inside of this method, the quantum function returns incorrect values or if
                 both continuous and discrete operations are specified in the same quantum circuit
         """
-        # pylint: disable=attribute-defined-outside-init, too-many-branches
+        # pylint: disable=attribute-defined-outside-init, too-many-branches, too-many-statements
 
-        # flatten the args, replace each argument with a Variable instance carrying a unique index
-        arg_vars = [Variable(idx) for idx, _ in enumerate(_flatten(args))]
-        self.num_variables = len(arg_vars)
-        # arrange the newly created Variables in the nested structure of args
-        arg_vars = unflatten(arg_vars, args)
+        arg_vars, kwarg_vars = self._make_variables(args, kwargs)
```

Comment:
Oh, I suppose the _flatten that used to be here has just moved into _make_variables :laughing: